src/AutoRest.CSharp/Common/AutoRest/Plugins/GeneratedCodeWorkspace.cs

```diff
@@ -91,6 +110,34 @@ private async Task<Document> ProcessDocument(Document document)
             return document;
         }
 
+        private bool ContainsSuppressedType(SyntaxTree syntaxTree, SemanticModel semanticModel)
+        {
+            var typeDeclarationSyntax = syntaxTree.GetRoot().DescendantNodes()
+                .OfType<BaseTypeDeclarationSyntax>()
+                .FirstOrDefault();
+
+            if (typeDeclarationSyntax == null)
+            {
+                return false;
+            }
+
+            var typeSymbol = semanticModel.GetDeclaredSymbol(typeDeclarationSyntax);
+            while (typeSymbol != null)
+            {
+                var fullName = typeSymbol.ToDisplayString(_fullyQualifiedNameFormat);
+                if (_suppressedTypeNames.Contains(typeSymbol.Name) || _suppressedTypeNames.Contains(fullName))
```

Comment:
Should we do equals?

Reply:
Equals what to what?

Reply:
My bad, though we were doing stirng.Contains.